src/plugins/str.rs

```diff
@@ -1,20 +1,22 @@
 use nu::{
     serve_plugin, CallInfo, Plugin, Primitive, ReturnSuccess, ReturnValue, ShellError, Signature,
-    SyntaxShape, Tagged, TaggedItem, Value,
+    SyntaxShape, Tagged, Tag, TaggedItem, Value,
 };
 
 #[derive(Debug, Eq, PartialEq)]
 enum Action {
     Downcase,
     Upcase,
     ToInteger,
+    Substring,
 }
 
 pub type ColumnPath = Vec<Tagged<String>>;
 
 struct Str {
     field: Option<ColumnPath>,
-    params: Option<Vec<String>>,
+    // I believe this should now support "named properties", like --substring "x,y"...
+    params: Vec<Tagged<String>>,
```

Comment:
If you store the substring start/finish positions in the enum, I think you don't have to do this step